### ISSUE TITLE
TDX patchset for 6.8 kernel

### ISSUE THREAD

**Issue by BastienFaivre** [canonical/tdx#113]
Hi,
I am trying to get the required host kernel for release 2.0. I saw on the release notes the link to [-intel kernel](https://git.launchpad.net/~canonical-kernel/ubuntu/+source/linux-intel/+git/noble/log/?h=main), but does this source code include the TDX v19 KVM patchset mentioned? I am asking because, when I try to set the kernel configuration (`make menuconfig`), some TDX parameters (such as `CONFIG_INTEL_TDX_HOST`) are missing (in the `.config` file).
So, if it is the case that the linked source code does not include the TDX patches, where can I find them?
To generalize the question, how do I get this required kernel?
Thank you in advance for your help!

**Comment by hector-cao** [comment]
Hello @BastienFaivre , yes this source code include the TDX v19 KVM patchset, you can see it in the git log history. Ubuntu has a special way to build the kernel and generate the kernel config, you can see the variable you are looking for here : https://git.launchpad.net/~canonical-kernel/ubuntu/+source/linux-intel/+git/noble/tree/debian.intel/config/annotations?h=main#n80

**Comment by BastienFaivre** [comment]
Hi @hector-cao, thank you for your response and the pointer to these variables. You mention a _special way_ to build the kernel, what is it? How do I include this `annotations' file in the kernel config? Is there any documentation about the kernel config generation and compilation somewhere?

**Comment by hector-cao** [comment]
Hi, 
Here are the instructions to build kernel for Ubuntu : https://wiki.ubuntu.com/Kernel/BuildYourOwnKernel

You can find more details about config annotations but it should not affect the way you build the ubuntu kernel:
https://discourse.ubuntu.com/t/kernel-configuration-in-ubuntu/35857/5

**Comment by BastienFaivre** [comment]
Hello, I was able to generate the config file and compile the kernel. Thank you for your help!
In case it would be useful for other people, here is what I did (which is different from the documentation):
1. Generate the config file: `./debian/scripts/misc/annotations --arch amd64 --export > .config`
2. Adjust the parameter for your needs: `make menuconfig`
3. Compile the kernel. Example with _deb_ packages: `make -j$(nproc) bindeb-pkg`
